src/shoppinglist.h

```diff
@@ -0,0 +1,158 @@
+	Q_OBJECT
+	public:
+		ShoppingList(QObject *parent = NULL);
+
+		/**
+		 * Get total price of all items in shopping list.
+		 **/
+		double getTotalPrice();
+
+		/**
+		 * \defgroup TableModelFunctions QAbstractTableModel subclassing
+		 * Functions related to the QAbstractTableModel subclassing.
+		 **/
+
+		/**
+		 * \ingroup TableModelFunctions
+		 * Number of rows in the table model represented by ShoppingList. Necessary for QAbstractTableModel subclassing.
+		 *
+		 * \param parent Parent index
+		 * \return Number of items in shopping list
+		 **/
+		virtual int rowCount(const QModelIndex &parent = QModelIndex()) const;
+
+		/**
+		 * \ingroup TableModelFunctions
+		 * Number of columns in the table model. Necessary for QAbstractTableModel subclassing.
+		 *
+		 * \param parent Parent index
+		 * \return Always 3 (name + price + amount)
+		 **/
+		virtual int columnCount(const QModelIndex &parent = QModelIndex()) const;
+
+		/**
+		 * \ingroup TableModelFunctions
+		 * Data access function. Necessary for QAbstractTableModel subclassing. Implements only Qt::DisplayRole.
+		 *
+		 * \param index Index in table. row is assumed to correspond to position in the shopping list, while columns 0, 1 and 2 correspond to name, price and amount (see also ITEM_NAME_COL etc. defined in shoppinglist.cpp)
+		 * \param role Role, see Qt docs for QAbstractItemModel
+		 * \return Name, price, amount or QVariant(QVariant::Invalid) depending on column index
+		 **/
+		virtual QVariant data(const QModelIndex &index, int role = Qt::DisplayRole) const;
+
+		/**
+		 * \ingroup TableModelFunctions
+		 * Data setting function. Necessary for QAbstractTableModel subclassing. Implements only Qt::EditRole, allows editing only for the amount of an already inserted item.
+		 *
+		 * \param index Index in table, see documentation for data()
+		 * \param value New value for the specified table position
+		 * \param role Role
+		 **/
+		virtual bool setData(const QModelIndex &index, const QVariant &value, int role = Qt::EditRole);
+
+		/**
+		 * \ingroup TableModelFunctions
+		 * Item flags for the defined table indices.
+		 *
+		 * \param index Index, see documentation for data()
+		 * \return Enabled and editable if the index corresponds to an amount for a shopping list item, disabled otherwise
+		 **/
+		virtual Qt::ItemFlags flags(const QModelIndex &index) const;
+	public slots:
+		/**
+		 * Add new item to shopping list. Item names function as unique identifiers:
+		 * 	- If the price is the same: Assumed that the item is the same, the input amount is
+		 * 	  added to the existing item amount
+		 * 	- If the price is different: This should happen, so the new item is ignored. :-P
+		 *
+		 * \param itemName Name of item (unique)
+		 * \param price Item price
+		 * \param amount Item amount (number of items, is accumulated to the item each time the SLOT is called)
+		 **/
+		void newItem(QString itemName, double price, int amount = 1);
```

Comment:
The cleanest solution is to require the price menu to be responsible for giving unique names to all of its items. The shopping list will be weird and complex if it has to handle a special generic item in a special way. Requiring the price menu to handle its generic items in a special way is well-defined, however, since it will have to handle an generic item in a special way anyway.

Wontfix'd. o:-)